src/WebJobs.Script/Extensions/FunctionMetadataExtensions.cs

```diff
@@ -89,11 +91,29 @@ internal static void SetFunctionId(this FunctionMetadata metadata, string functi
         public static void SetIsCodeless(this FunctionMetadata metadata, bool value) =>
             metadata.Properties[IsCodelessKey] = value;
 
+        public static void SetRetry(this FunctionMetadata metadata, RetryOptions value) =>
+            metadata.Properties[RetryKey] = value;
+
+        public static RetryOptions GetRetry(this FunctionMetadata metadata) =>
+            GetObjectProperty(metadata.Properties, RetryKey) as RetryOptions;
+
         private static bool GetBoolProperty(IDictionary<string, object> properties, string propertyKey)
         {
             return properties.TryGetValue(propertyKey, out object valObj)
                 && valObj is bool valBool
                 && valBool;
         }
+
+        private static object GetObjectProperty(IDictionary<string, object> properties, string propertyKey)
```

Comment:
What value is this method providing? The underlying dictionary already returns objects, so I'd just code directly to the dictionary in GetRetry above